<launch>
  <arg name = "destination" default = "192.168.0.1" />
  <arg name = "port_send" default = "54650" />
  <arg name = "port_receive" default = "54630" /> 
  <arg name = "connection_type" default = "rmpc" />
  <arg name = "direct_control_flag" default = "false" />

  <arg name = "car_info_topic" default = "/comsctrl_car_info" />
  <arg name = "twist_cmd_topic" default = "/twist_cmd" />
  <arg name = "accel_cmd_topic" default = "/accel_cmd" />
  <arg name = "steer_cmd_topic" default = "/steer_cmd" />
  <arg name = "ndt_pose_topic" default = "/ndt_pose" />
  <arg name = "ndt_stat_topic" default = "/ndt_stat" />
  <arg name = "estimate_velocity_topic" default = "/estimate_velocity" />


  <node pkg="udp_converter" type="comsctrl_converter" name ="comsctrl_converter" output="screen">
    <param name = "direct_control_flag" value = "$(arg direct_control_flag)" />
    <param name = "connection_type" value = "$(arg connection_type)" />
    <param name = "car_info_topic" value = "$(arg car_info_topic)" />
    <param name = "twist_cmd_topic" value = "$(arg twist_cmd_topic)" />
    <param name = "accel_cmd_topic" value = "$(arg accel_cmd_topic)" />
    <param name = "steer_cmd_topic" value = "$(arg steer_cmd_topic)" />
    <param name = "ndt_pose_topic" value = "$(arg ndt_pose_topic)" />
    <param name = "ndt_stat_topic" value = "$(arg ndt_stat_topic)" />
    <param name = "estimate_velocity_topic" value = "$(arg estimate_velocity_topic)" />
  </node>
  
  <node pkg = "udp_connector" type = "comsctrl_sender" name = "comsctrl_sender">
    <param name = "destination" value = "$(arg destination)" />
    <param name = "port_send" value = "$(arg port_send)" />
    <param name = "connection_type" value = "$(arg connection_type)" />
  </node>

  <node pkg = "udp_connector" type = "comsctrl_receiver" name = "comsctrl_receiver">
    <param name = "port_receive" value = "$(arg port_receive)" />
    <param name = "connection_type" value = "$(arg connection_type)" />
  </node>
</launch>

